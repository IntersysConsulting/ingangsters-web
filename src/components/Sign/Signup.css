svg {
  color: var(--white);
}

.btn-intersys {
  color: var(--white);
  background-color: var(--green);
}
.btn-intersys:hover {
  color: var(--green);
}

.link {
  color: var(--light-blue);
}

.signup-form {
  background-color: var(--light-grey);
}

.icon {
  color: white;
  cursor: pointer;
}

.icon:hover {
  color: var(--light-blue);
  cursor: pointer;
}

h2,h5 {
  color: var(--light-blue);
}
.form-title {
  color: var(--light-blue);
  border-bottom: 2px solid var(--light-blue);
}
