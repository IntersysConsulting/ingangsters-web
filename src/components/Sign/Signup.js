import React, { useState } from "react";
import SimpleNavBar from "../SimpleNavBar/SimpleNavBar";
import { Redirect, Link } from "react-router-dom";
import "./Sigup.css";
import "../../css/colors.css";
import { FaEye } from "react-icons/fa";
import { connect } from "react-redux";
import { login } from "../../actions/creators/auth";
import { Form, Row, Col } from "react-bootstrap";

const Login = ({ login, isAuthenticated }) => {
  const [values, setValues] = useState({
    email: "",
    password: ""
  });

  const [showPassword, setShowPassword] = useState(false);
  const [showPasswordConfirm, setShowPasswordConfirm] = useState(false);

  const { firstName, lastName,phone, email, password,passwordConfirm } = values;

  const handleChange = name => event => {
    setValues({ ...values, error: false, [name]: event.target.value });
  };

  const onSubmit = async event => {
    event.preventDefault();
    login(email, password);
  };

  if (isAuthenticated) {
    return <Redirect to="/" />;
  }

  const toggleShow = () => {
      setShowPassword(!showPassword);
  };
  const toggleShowConfirm = () => {
    setShowPasswordConfirm(!showPasswordConfirm);
};

  const signUpForm = () => (
    <div className="container mt-5">
      <div className="row login-form">
        <div className="col-12">
          <h2>Sign up</h2>
          <h5>Please fill this form to create an account! </h5>
          <hr />
          <form>
            <div className="form-group">
              <Row>
                <Col>
                  <label className="text-muted">First name</label>
                  <input
                    onChange={handleChange("firstName")}
                    type="text"
                    placeholder="First name"
                    className="form-control"
                    value={firstName}
                  />
                </Col>
                <Col>
                  <label className="text-muted">Last name</label>
                  <input
                    onChange={handleChange("lastName")}
                    type="text"
                    placeholder="Last name"
                    className="form-control"
                    value={lastName}
                  />
                </Col>
              </Row>
            </div>

            <div className="form-group">
              <label className="text-muted">Phone</label>
              <input
                onChange={handleChange("phone")}
                type="tel"
                pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
                placeholder="Phone (Optional)"
                className="form-control"
                value={phone}
              />
            </div>

            <div className="form-group">
              <label className="text-muted">Password</label>
              <div className="row">
                <div className="col-10 col-sm-10 col-md-11 col-lg-11 pr-0">
                  <input
                    onChange={handleChange("password")}
                    type={showPassword ? "text" : "password"}
                    placeholder="password"
                    className="form-control"
                    value={password}
                  />
                </div>
                <div className="col-1 text-center" onClick={toggleShow}>
                  <FaEye size={32} />
                </div>
              </div>
            </div>

            <div className="form-group">
              <label className="text-muted">Confirm Password</label>
              <div className="row">
                <div className="col-10 col-sm-10 col-md-11 col-lg-11 pr-0">
                  <input
                    onChange={handleChange("passwordConfirm")}
                    type={showPasswordConfirm ? "text" : "password"}
                    placeholder="Confirm password"
                    className="form-control"
                    value={passwordConfirm}
                  />
                </div>
                <div className="col-1 text-center" onClick={toggleShowConfirm}>
                  <FaEye size={32} />
                </div>
              </div>
            </div>





            <p className="link">Forgot password</p>
            <div className="mb-4 text-center">
              <button
                onClick={onSubmit}
                className="btn btn-intersys btn-block my-3"
              >
                Login
              </button>
              <Link className="link" to="/signup">
                Create an account
              </Link>
            </div>
          </form>
        </div>
      </div>
    </div>
  );

  return (
    <div>
      <SimpleNavBar />
      {signUpForm()}
    </div>
  );
};

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated
});

export default connect(
  mapStateToProps,
  { login }
)(Login);
