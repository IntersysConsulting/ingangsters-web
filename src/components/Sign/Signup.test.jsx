import React from "react";
import Adapter from "enzyme-adapter-react-16";
import Enzyme, { shallow } from "enzyme";
import Signup from "./Signup";

Enzyme.configure({ adapter: new Adapter() });

describe("SignUp Form", () => {
  it("SignUp Form renders correctly", () => {
    const component = shallow(<Signup />);
    expect(component).toMatchSnapshot();
  });
});