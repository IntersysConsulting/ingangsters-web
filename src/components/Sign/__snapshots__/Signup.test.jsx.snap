// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Form SignUp Form renders correctly 1`] = `
<div>
  <SimpleNavBar />
  <Formik
    enableReinitialize={false}
    initialValues={
      Object {
        "email": "",
        "firstName": "",
        "lastName": "",
        "password": "",
        "passwordConfirm": "",
        "phone": "",
      }
    }
    isInitialValid={false}
    onSubmit={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={
      ObjectSchema {
        "_blacklist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "_conditions": Array [],
        "_defaultDefault": [Function],
        "_deps": Array [],
        "_excludedEdges": Array [],
        "_exclusive": Object {},
        "_mutate": undefined,
        "_nodes": Array [
          "terms",
          "passwordConfirm",
          "password",
          "email",
          "phone",
          "lastName",
          "firstName",
        ],
        "_options": Object {
          "abortEarly": true,
          "recursive": true,
        },
        "_type": "object",
        "_typeError": [Function],
        "_whitelist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "fields": Object {
          "email": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": true,
              "undefined": false,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "firstName": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "lastName": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "password": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "min": true,
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "passwordConfirm": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "min": true,
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {
                null,
              },
              "refs": Map {
                "password" => Reference {
                  "getter": [Function],
                  "isContext": false,
                  "isSibling": true,
                  "isValue": false,
                  "key": "password",
                  "map": undefined,
                  "path": "password",
                },
              },
            },
            "_whitelistError": [Function],
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "phone": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "undefined": false,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "terms": BooleanSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "boolean",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
      }
    }
  >
    <Component />
  </Formik>
</div>
`;
